### ISSUE TITLE
Unable to bind contextmanager methods inside classmethods

### ISSUE THREAD

**Issue by Gobot1234** [microsoft/pyright#5530]
**Describe the bug**
pyright is unable to "bind" contextmanager methods inside of classmethods with a strange error

**To Reproduce**
```python
from contextlib import contextmanager


class Foo:
    @classmethod
    def foo(cls) -> None:
        cls.open

    @contextmanager
    def open(self):
        yield
```
**Expected behavior**
No error to be produced

**Code or Screenshots**
<img width="837" alt="image" src="https://github.com/microsoft/pyright/assets/50501825/2770a9e7-bb4b-4c82-a66d-fd92b978756f">

**VS Code extension or command-line**
pylance: v2023.7.21

**Comment by cainmagi** [comment]
Take a look at this:
```python
from contextlib import contextmanager
from typing_extensions import reveal_type


class Foo:
    @classmethod
    def foo(cls) -> None:
        with cls.open() as ctx:
            reveal_type(ctx)  # None, as expected.

    @contextmanager
    @classmethod
    def open(cls):
        yield
```

**Comment by Gobot1234** [comment]
I don't particularly see how that's relevant here, I'm trying to access cls.open as an unbound method, I don't care to call it or change it to be a classmethod.

**Comment by erictraut** [comment]
@cainmagi, I appreciate your attempts to contribute to the pyright forums, but several of your recent answers have generated more confusion than help. Unless you're really confident in your answer, please let me or someone else answer to avoid such confusion.

**Comment by erictraut** [comment]
Thanks for the bug report. This was a very old bug — probably around since the beginning of pyright. It will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.318, which I just published. It will also be included in a future release of pylance.
